src/CPUSolver.cpp

```diff
@@ -490,27 +504,43 @@ void CPUSolver::setupMPIBuffers() {
               (dx == 0 && dy == 0 && dz != 0)) {
             int domain = _geometry->getNeighborDomain(dx, dy, dz);
             if (domain != -1) {
-              neighbor_connections.insert({domain, idx});
-              float* send_buffer = new float[length];
-              _send_buffers.push_back(send_buffer);
-              float* receive_buffer = new float[length];
-              _receive_buffers.push_back(receive_buffer);
+              _neighbor_connections.insert({domain, idx});
               _neighbor_domains.push_back(domain);
               idx++;
+
+              /* Inititalize vector of starting indexes into send_buffers */
+              _send_buffers_index.push_back(0);
             }
           }
         }
       }
     }
 
-    /* Setup Track communication information for all neighbor domains */
+    /* Estimate and print size of flux transfer buffers */
+    int size_mb = 2 * message_length * _send_buffers.size() * sizeof(float);
+    int max_size_mb;
+    MPI_Allreduce(&size_mb, &max_size_mb, 1, MPI_INT, MPI_MAX,
+                  _geometry->getMPICart());
+    log_printf(NORMAL, "Max track fluxes transfer buffer storage = %.2f MB",
+               max_size_mb / 1e6);
```

Comment:
line 519~525 should be put after line 535. otherwise, size_mb is zero

Reply:
Good catch. Not sure why it wasnt printing 0 for me